compiler/rustc_attr/src/builtin.rs

```diff
@@ -658,8 +654,10 @@ where
             continue;
         }
 
-        if depr.is_some() {
-            struct_span_err!(diagnostic, item_sp, E0550, "multiple deprecated attributes").emit();
+        if let Some((_, span)) = &depr {
+            struct_span_err!(diagnostic, attr.span, E0550, "multiple deprecated attributes")
+                .span_note(*span, "first deprecation attribute here")
```

Comment:
Can this happen multiple times (if you have three deprecation attributes)? If so, reword the label to not make it technically incorrect.

Also, use prefer to use `span_label` when possible.

```suggestion
            struct_span_err!(diagnostic, attr.span, E0550, "multiple deprecated attributes")
                .span_label(attr.span, "repeated deprecated attribute")
                .span_label(*span, "prior deprecation attribute")
```

Reply:
It stops right after the first duplicate, so having three deprecation attributes wouldn't output any more diagnostics than two.

Is there documentation somewhere about all the different options for diagnostics, like `span_label`, `span_suggestion_short`, etc.?

Reply:
https://github.com/rust-lang/rust/blob/master/compiler/rustc_errors/src/diagnostic.rs has all the methods that can be used and they have good docstrings for the most part.